### PR TITLE
[video_player_avplay] Add setStreamingProperty API

### PR DESCRIPTION
**Pull request by xiaowei-guan**
Add setStreamingProperty API:

```dart
  /// Sets specific feature values for HTTP, MMS, or specific streaming engine (Smooth Streaming, HLS, DASH, DivX Plus Streaming, or Widevine).
  /// The available streaming properties depend on the streaming protocol or engine.
  /// Use the CUSTOM_MESSAGE property for streaming engine or CP-specific settings.
  Future<void> setStreamingProperty(StreamingPropertyType type, String value) ;
```
Native interface:
```C++
/**
   * @brief     Set streamingengine property
   * @param     [in] type : attribute type. \n
   *             type can be \n
   *               "COOKIE" \n
   *               "USER_AGENT" \n
   *               "RESUME_TIME" \n
   *               "ADAPTIVE_INFO"
   * @param     [in] value : value of attribute type
   * @pre       The player state can be all of #State except #State::kNone
   */
  virtual void SetStreamingProperty(const std::string& type,
                                    const std::string& value)
```

### CODE REVIEW

File: packages/video_player_avplay/lib/video_player.dart [hunk 1]
```diff
@@ -712,6 +712,17 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     return _videoPlayerPlatform.getStreamingProperty(_playerId, type);
   }
 
+  /// Sets specific feature values for HTTP, MMS, or specific streaming engine (Smooth Streaming, HLS, DASH, DivX Plus Streaming, or Widevine).
+  /// The available streaming properties depend on the streaming protocol or engine.
+  /// Use the CUSTOM_MESSAGE property for streaming engine or CP-specific settings.
```
Comment (JSUYA) [inline packages/video_player_avplay/lib/video_player.dart:717]: Just curious, I see [this description](https://developer.samsung.com/smarttv/develop/api-references/samsung-product-api-references/avplay-api.html?device=htv) is in the avplay document, but is this actually a feature we can support?

File: packages/video_player_avplay/tizen/src/plus_player.cc [hunk 2]
```diff
@@ -96,8 +96,8 @@ int64_t PlusPlayer::Create(const std::string &uri,
   if (create_message.streaming_property() != nullptr &&
       !create_message.streaming_property()->empty()) {
     for (const auto &[key, value] : *create_message.streaming_property()) {
-      SetStreamingProperty(player_, std::get<std::string>(key),
-                           std::get<std::string>(value));
+      SetStreamingProp(std::get<std::string>(key),
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply packages/video_player_avplay/tizen/src/plus_player.cc]: Done

File: packages/video_player_avplay/tizen/src/plus_player.cc [hunk 3]
```diff
@@ -597,6 +597,23 @@ bool PlusPlayer::SetBufferConfig(const std::string &key, int64_t value) {
...
 
+void PlusPlayer::SetStreamingProp(const std::string &type,
+                                  const std::string &value) {
+  if (!player_) {
+    LOG_ERROR("[PlusPlayer] Player not created.");
+    return;
+  }
+  plusplayer::State state = GetState(player_);
+  if (state == plusplayer::State::kNone) {
+    LOG_ERROR("[PlusPlayer]:Player is in invalid state[%d]", state);
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply packages/video_player_avplay/tizen/src/plus_player.cc]: Done

File: packages/video_player_avplay/lib/video_player.dart (hunk 1 above)
Comment (xiaowei-guan) [reply packages/video_player_avplay/lib/video_player.dart:717]: The Web avplay plugin and flutter avplay  plugin both based on plusplayer, if Web avplay plugin can support, we can also support it.

**Review by JSUYA — APPROVED** [review]

**Pull request merged**